src/audio-manager.ts

```diff
@@ -0,0 +1,208 @@
+        const unlockHandler = () => {
+            _audioContext.resume().then(() => {
+                window.removeEventListener('click', unlockHandler);
+                window.removeEventListener('touch', unlockHandler);
+                window.removeEventListener('keydown', unlockHandler);
+                window.removeEventListener('mousedown', unlockHandler);
+                resolve();
+            });
+        };
+
+        window.addEventListener('click', unlockHandler);
+        window.addEventListener('touch', unlockHandler);
+        window.addEventListener('keydown', unlockHandler);
+        window.addEventListener('mousedown', unlockHandler);
+    });
+}
+
+/**
+ * Represents a playable audio node that can be used to play audio panned or without panning.
+ *
+ * @note Use the `destroy()` method if audio is not going to be used anymore, to avoid unused audio files
+ * clogging up memory
+ */
+class PlayableNode {
+    /** Whether to loop the audio */
+    public loop: boolean = false;
+
+    /** Whether to enable HRTF over regular panning */
+    public HRTF: boolean = false;
+
+    private _audioBuffer: AudioBuffer;
+    private _source: string;
+    private _isPlaying: boolean = false;
+    private _gainNode: GainNode = new GainNode(_audioContext);
+    private _pannerNode: PannerNode | undefined;
+    private _audioNode: AudioBufferSourceNode = new AudioBufferSourceNode(_audioContext);
+    private _destroy: boolean = false;
+
+    constructor(src: string, audioBuffer: AudioBuffer) {
+        this._audioBuffer = audioBuffer;
+        this._source = src;
+        this._gainNode.connect(_audioContext.destination);
+    }
+
+    /**
+     * Asynchronously plays the loaded audio. If the audio is already playing, it stops the current playback and starts anew.
+     * If the audio context is in a suspended state, it attempts to unlock the audio context before playing and will
+     * continue after the user has interacted with the website.
+     *
+     * @param posVec - An optional parameter representing the 3D spatial position of the audio source.
+     *                 If provided, the audio will be spatialized using a PannerNode based on the given position vector.
+     * @returns A Promise that resolves once the audio playback starts.
+     * @throws - If there is an error during the playback process, a warning is logged to the console.
+     */
+    async play(posVec?: Float32Array): Promise<void> {
+        try {
+            if (this.isPlaying) {
+                this.stop();
+            }
+            if (_audioContext.state === 'suspended') {
+                await _unlockAudioContext();
+            }
+            this._audioNode = new AudioBufferSourceNode(_audioContext, {
+                buffer: this._audioBuffer,
+                loop: this.loop,
+            });
+            if (posVec !== undefined) {
+                this._pannerNode = new PannerNode(_audioContext, {
+                    coneInnerAngle: 360,
```

Comment:
The user will never need to control those settings?

Reply:
Hmm, i've added an optional parameter to configure the entire thing. But i am unsure if that's the correct way to do it.